examples/cluster/plot_kmeans_assumptions.py

```diff
@@ -0,0 +1,73 @@
+"""
+====================================
+Demonstration of k-means assumptions
+====================================
+
+This example is meant to illustrate situations where k-means will produce good
+clusters and situations where k-means will go wrong. In the first three plots,
+k-means produces incorrect clusters because it makes assumptions about the data
+that turn out to be incorrect. In the last plot, k-means correctly clusters data
+despite unbalanced blobs.
+"""
+print(__doc__)
+
+# Author: Phil Roth <mr.phil.roth@gmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from sklearn.cluster import KMeans
+from sklearn.datasets import make_blobs
+
+plt.figure(figsize=(12, 12))
+
+np.random.seed(170)
```

Comment:
We prefer using random state objects and not altering global state. Global state might change behavior when building the documentation.